lms/djangoapps/lti_provider/tests/test_outcomes.py

```diff
@@ -366,3 +433,279 @@ def test_response_with_failing_status_field(self):
+        self.create_graded_assignment(self.vertical_desc, 'graded_vertical', self.outcome_service)
+        assignments = outcomes.get_assignments_for_problem(
+            self.unit_desc, self.user_id, self.course_key
+        )
+        self.assertEqual(len(assignments), 2)
+        self.assertEqual(len(assignments[self.unit_desc]), 1)
+        self.assertEqual(len(assignments[self.vertical_desc]), 1)
+        self.assertEqual(
+            assignments[self.unit_desc][0].lis_result_sourcedid,
+            'graded_unit'
+        )
+        self.assertEqual(
+            assignments[self.vertical_desc][0].lis_result_sourcedid,
+            'graded_vertical'
+        )
+
+    def test_with_unit_used_twice(self):
+        self.create_graded_assignment(self.unit_desc, 'graded_unit', self.outcome_service)
+        self.create_graded_assignment(self.unit_desc, 'graded_unit2', self.outcome_service)
+        assignments = outcomes.get_assignments_for_problem(
+            self.unit_desc, self.user_id, self.course_key
+        )
+        self.assertEqual(len(assignments), 1)
+        self.assertEqual(len(assignments[self.unit_desc]), 2)
+        self.assertEqual(
+            assignments[self.unit_desc][0].lis_result_sourcedid,
+            'graded_unit'
+        )
+        self.assertEqual(
+            assignments[self.unit_desc][1].lis_result_sourcedid,
+            'graded_unit2'
+        )
+
+    def test_with_unit_graded_for_different_user(self):
+        self.create_graded_assignment(self.unit_desc, 'graded_unit', self.outcome_service)
+        other_user = UserFactory.create()
+        assignments = outcomes.get_assignments_for_problem(
+            self.unit_desc, other_user.id, self.course_key
+        )
+        self.assertEqual(len(assignments), 0)
+
+    def test_with_unit_graded_for_multiple_consumers(self):
+        other_outcome_service = self.create_outcome_service('second_consumer')
+        self.create_graded_assignment(self.unit_desc, 'graded_unit', self.outcome_service)
+        self.create_graded_assignment(self.unit_desc, 'graded_unit2', other_outcome_service)
+        assignments = outcomes.get_assignments_for_problem(
+            self.unit_desc, self.user_id, self.course_key
+        )
+        self.assertEqual(len(assignments), 1)
+        self.assertEqual(len(assignments[self.unit_desc]), 2)
+        self.assertEqual(
+            assignments[self.unit_desc][0].lis_result_sourcedid,
+            'graded_unit'
+        )
+        self.assertEqual(
+            assignments[self.unit_desc][1].lis_result_sourcedid,
+            'graded_unit2'
+        )
+        self.assertEqual(
+            assignments[self.unit_desc][0].outcome_service,
+            self.outcome_service
+        )
+        self.assertEqual(
+            assignments[self.unit_desc][1].outcome_service,
+            other_outcome_service
+        )
+
+
+class TestCalculateScore(TestCase):
```

Comment:
Generally, this test file is getting rather long. I'd encourage you to split it into smaller files.